polars/polars-io/src/csv/write_impl.rs

```diff
@@ -195,42 +197,35 @@ pub(crate) fn write<W: Write>(
         .map_err(|_| PolarsError::ComputeError("quote char leads invalid utf8".into()))?;
     let delimiter = char::from(options.delimiter);
 
-    // if datetime format not specified, infer the maximum required precision
+    let mut formats: Vec<Option<&str>> = vec![];
```

Comment:
If we `iter().map( the implementation ).collect::<Vec<_>()` the Vec will be preallocated to the proper length.